scripts/deploy.py

```diff
@@ -337,6 +339,72 @@ def check_breakage(app_name, current_release_version):
                 'version.')
 
 
+def check_travis_and_circleci_tests(current_branch_name):
+    """Checks if all travis and circleci tests are passing on release branch.
+
+    Args:
+        current_branch_name: str. The name of current branch.
+
+    Raises:
+        Exception: The latest commit on release branch locally does not match
+            the latest commit on local fork or upstream.
+        Exception: The travis or circleci local instance is not setup.
+        Exception: The travis or circleci tests are failing.
+    """
+    local_sha = subprocess.check_output([
+        'git', 'rev-parse', current_branch_name])
+    origin_sha = subprocess.check_output([
+        'git', 'rev-parse', 'origin/%s' % current_branch_name])
+    upstream_sha = subprocess.check_output([
+        'git', 'rev-parse', '%s/%s' % (
+            common.get_remote_alias(release_constants.REMOTE_URL),
+            current_branch_name)])
+    if local_sha != origin_sha:
+        raise Exception(
+            'The latest commit on release branch locally does '
+            'not match the latest commit on your local fork.')
+    if local_sha != upstream_sha:
+        raise Exception(
+            'The latest commit on release branch locally does '
+            'not match the latest commit on Oppia repo.')
+
+    python_utils.PRINT('Enter your GitHub username.\n')
+    github_username = python_utils.INPUT().lower()
+
+    travis_url = 'https://travis-ci.org/%s/oppia/branches' % github_username
+    circleci_url = 'https://circleci.com/gh/%s/workflows/oppia' % (
+        github_username)
+
+    if requests.get(travis_url).status_code != 200:
+        raise Exception(
+            'Please setup local travis instance before deploying: https:'
+            '//github.com/oppia/oppia/wiki/Setup-your-own-Travis-instance')
+
+    if requests.get(circleci_url).status_code != 200:
+        raise Exception(
```

Comment:
I thought I left a comment here last review, but apparently I hadn't.... Do we expect the deployer to check on a local instance of travis/circleCI always? Why can't we use the status of the build that happens on oppia/release-x.y.z branch? (AFAIK it automatically triggers a build for every push, cherrypick, etc).

Reply:
Ah I see. Hmmm, Maybe it would help with the long queue issue. I don't know if it is a good idea to make this mandatory. Maybe better to have it optional to have your own instance but at least the tests pass on Oppia/release-x.y.z branch?

Reply:
Is there any way for us to verify that the build on the separate instance uses the updated version of the release branch? If not I would rather have it on the main repo branch.

Reply:
Yes, I have checked it in this test itself (I have matched the sha of last commit with upstream and origin sha). 

I think it's a good idea to not make it mandatory to check on local instance of travis/circleci. I have made the updates for the same.